lib/index.js

```diff
@@ -370,14 +426,28 @@ exports.getStorageClient = (ctx) => {
     const url = `${ctx.configuration.fusebit_storage_audience}/v1/account/${ctx.configuration.fusebit_storage_account_id}/subscription/${ctx.configuration.fusebit_storage_subscription_id}/storage/${ctx.configuration.fusebit_storage_id}`;
 
     return {
+        etag: null,
         get: async () => {
             const response = await Superagent.get(url)
                 .set('Authorization', `Bearer ${accessToken}`)
                 .ok((res) => res.status < 300 || res.status === 404);
+            this.etag = response.headers.etag;
             return response.status === 404 ? undefined : response.body.data;
         },
-        put: async (data) => {
-            await Superagent.put(url).set('Authorization', `Bearer ${accessToken}`).send({ data });
+        put: async (data, oldEtag) => {
+            let request = Superagent.put(url).set('Authorization', `Bearer ${accessToken}`);
+            if (oldEtag){
+              request = request.set('If-Match', oldEtag);
+            }
+            const response = await request.send({ data });
+            if(response.statusCode == 412){
```

Comment:
Do we really return 412? It should have been 409.

Reply:
Actually there was no need to really re-wrap the exception here, so I removed this code path and let it flow through as is. Also it is actually a 409 like you said, so this code path never even worked.